### ISSUE TITLE
Adaptive activations

### ISSUE THREAD

**Issue by kevinmartinmauro** [fchollet/keras#3455]
Is it possible to perform adaptive activations in Keras? See: http://citeseerx.ist.psu.edu/showciting?cid=1882454

My understanding is that a set of neurons can serve as the weight-constants for a polynomial activation function.

 I.e. Y = N1_x^2_+N2_x_+N3, where N is a neuron

The activation function changes as training progresses to better serve its task

**Comment by ncullen93** [comment]
yes look at keras.layers.advanced_activations - for instance, PReLU. Simply set 'self.trainable_weights' in the 'build' method of the class, and use those weights in the 'call' method.

class PReLU(Layer):
    '''Parametric Rectified Linear Unit:
    `f(x) = alphas * x for x < 0`,
    `f(x) = x for x >= 0`,
    where `alphas` is a learned array with the same shape as x.

    # Input shape
        Arbitrary. Use the keyword argument `input_shape`
        (tuple of integers, does not include the samples axis)
        when using this layer as the first layer in a model.

    # Output shape
        Same shape as the input.

    # Arguments
        init: initialization function for the weights.
        weights: initial weights, as a list of a single Numpy array.

    # References
        - [Delving Deep into Rectifiers: Surpassing Human-Level Performance on ImageNet Classification](http://arxiv.org/pdf/1502.01852v1.pdf)
    '''
    def __init__(self, init='zero', weights=None, **kwargs):
        self.supports_masking = True
        self.init = initializations.get(init)
        self.initial_weights = weights
        super(PReLU, self).__init__(**kwargs)

    def build(self, input_shape):
        self.alphas = self.init(input_shape[1:],
                                name='{}_alphas'.format(self.name))
        self.trainable_weights = [self.alphas]

        if self.initial_weights is not None:
            self.set_weights(self.initial_weights)
            del self.initial_weights

    def call(self, x, mask=None):
        pos = K.relu(x)
        neg = self.alphas * (x - abs(x)) * 0.5
        return pos + neg

    def get_config(self):
        config = {'init': self.init.__name__}
        base_config = super(PReLU, self).get_config()
        return dict(list(base_config.items()) + list(config.items()))
